### ISSUE TITLE
Comment author must not be equal to existing wordpress user

### ISSUE THREAD

**Issue by tonimain** [wp-graphql/wp-graphql#172]
Right now we are returning inside comments {} a new authors {} type wich is right now related to an existing user inside wordpress.
But a comment author must not be registered in wordpress (get_comment_author) and could also exist as name, email type.

So as discussed with jason some approach would be to integrate a basic PersonType/UserType wich could be inside wordpress or non-wordpress.

**Comment by jasonbahl** [comment]
So, I'm thinking the best solution for this would be to make the `author` field on comments a union field, that can either return a user, or a guest author (non-user commenter).

Then you could do a query like this: 

```
comment(id: "...") {
  id 
  content
  author {
    ...on user {
      username
      userId
      ...otherFieldsForUsers
    }
    ...on guestAuthor {
      name
      email
      ip
      url
    }
  }
}
```

Need to think about this a bit more. 🤔

**Comment by jasonbahl** [comment]
This was addressed by #174

**Comment by jasonbahl** [comment]
Oops, closed the wrong issue. Re-opening.

**Comment by jasonbahl** [comment]
@pjpak Here's what I think needs to happen with this: 

We should make the `author` field on a comment return a `commentAuthorConnection` which has edges/node. 

The edge should contain the ipAddress that is stored under `wp_comments.comment_author_ip` and the node should be a union

The node on the connection should be a union that could return either a `user` or a `commentAuthor` (a new type we will need to define). 

The `commentAuthor` type should have the following fields: id, name, email and url – populating the data from the wp_comments.comment_author, comment_author_email and comment_author_url, and populating the `id` field from the comment_author_email using the same `Relay::toGlobalId()` we use elsewhere for global IDs.

Then, we should be able to do a query like: 

```
{
    comments {
        edges {
            node {
                id
                author {
                    edges {
                          cursor
                          ipAddress
                          node {
                              ...on user {
                                   id
                                   name
                                   email
                                   url
                              }
                              ...on commentAuthor {
                                  id
                                  name
                                  email
                                  url
                              }
                          }
                     }
                }
            }
        }
    }
}
```

By going this way, client side caches (such as Apollo) can normalize comment author nodes and can update that normalized cache much more efficiently. For example, let's say, hypothetically, a commentAuthor changed it's name. . .if properly normalized in a client side cache, the author would only be stored in the cache once and all instances rendering it would render from that one instance's data. . .so a name change to that one instance in the cache would cause all instances in the rendered page to update automatically!

This also allows the edge data, such as IP address to truly be represented as edge data. . .the IP address truly isn't a property of a comment or the user, it's a property of the connection between the two. . .the same commentAuthor could use their computer for one comment and their phone for another, so caching the IP address as a property of the author doesn't make sense and caching it as part of the comment also doesn't make sense, as comments themselves don't have IP addresses. . .Users have IP addresses, but they vary over time and should be cached only in the context (the edge) where they're relevant.

**Comment by jasonbahl** [comment]
This was addressed by #240
